HotelApp/DbContexts/Repositories/HotelRepository.cs

```diff
@@ -0,0 +1,72 @@
+﻿using HotelApp.API.DbContexts.Entities;
+using HotelApp.API.Models;
+using Microsoft.AspNetCore.Identity;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using AutoMapper;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.EntityFrameworkCore;
+
+namespace HotelApp.API.DbContexts.Repositories
+{
+    public class HotelRepository : IHotelRepository
+    {
+        private readonly HotelAppContext _hotelAppContext;
+        private readonly UserManager<User> _userManager;
+        private readonly IMapper _mapper;
+
+        public HotelRepository(HotelAppContext hotelAppContext,
+                               UserManager<User> userManager,
+                               IMapper mapper)
+        {
+            _hotelAppContext = hotelAppContext;
+            _userManager = userManager;
+            _mapper = mapper;
+        }
+
+        public async Task CreateHotelAsync(RegisterHotelDTO model, HotelStatus status, ClaimsPrincipal currentUser)
```

Comment:
Ovdje bi ostavio samo RegisterHotelDto kao ulazni parametar. Korisnika možeš dohvatit preko IHttpContextAccessor.HttpContext.User il tak nešto, a kod kreiranja hotela bi se status automatski trebao postavit u obradi ili kak se već zove.